### DISCUSSION TITLE
Cannot reference table directly?

### DISCUSSION THREAD

**Discussion by ckunte** [typst/typst#3449] [General]
### Description

> error: cannot reference table directly, try putting it into a figure

I get this above error when I label a table like so:

```typst
#table(
  columns: (1fr, auto, auto),
  inset: 10pt,
  align: horizon,
  [], [*North Sea*], [*South China Sea*],
  [Barge size (m)], [>76 $times$ >23], [91.4 $times$ 27.4],
  [Roll ($alpha$, $T_r$)], [20$deg$, 10s], [12.5$deg$, 5s],
  [Pitch ($beta$, $T_p$)], [12.5$deg$, 10s], [8$deg$, 5.5s],
  [Heave (gh)], [0.2g], [0.2g]
) <motion>
```
and call the referenced label like so:

```typst
where, $T_r$, $T_p$, and $T_h$ are full cycle periods associated with roll, 
pitch and heave respectively (in seconds). They are given in @motion.
```

Would referencing tables directly be possible in the near future?

### Use Case

I am trying to reference a table directly via a label, but it appears this is not (yet?) supported?

**Comment by laurmaedje** [comment]
It was an intentional design decision to require figure-like things that can have captions (figures, tables, code listing) to be an explicit figure. For that reason, tables also don't have their own caption mechanism. So, in your case, just write 
```typ
#figure(table(
  columns: (1fr, auto, auto),
  inset: 10pt,
  align: horizon,
  [], [*North Sea*], [*South China Sea*],
  [Barge size (m)], [>76 $times$ >23], [91.4 $times$ 27.4],
  [Roll ($alpha$, $T_r$)], [20$deg$, 10s], [12.5$deg$, 5s],
  [Pitch ($beta$, $T_p$)], [12.5$deg$, 10s], [8$deg$, 5.5s],
  [Heave (gh)], [0.2g], [0.2g]
)) <motion>
```
and then you can reference it as expected and it will say "Table 1".

**Comment by ckunte** [comment]
> It was an intentional design decision to require figure-like things that can have captions

Thank you for explaining this; much appreciate.
